lib/Clam/CfgBuilder.cc

```diff
@@ -279,7 +279,8 @@ static Optional<ref_cst_t> cmpInstToCrabRef(CmpInst &I, crabLitFactory &lfac,
 }
 
 /* If possible, return a Crab linear constraint from CmpInst */
-static Optional<lin_cst_t> cmpInstToCrabInt(CmpInst &I, crabLitFactory &lfac,
+static std::pair<Optional<lin_cst_t>, Optional<var_t>> cmpInstToCrabInt(CmpInst &I, crabLitFactory &lfac,
```

Comment:
I don't like to return a pair of things. 
Split
```
static Optional<lin_cst_t> cmpInstToCrabInt(CmpInst &I, crabLitFactory &lfac, const bool isNegated);
```

into

```
static Optional<lin_cst_t> signedCmpInstToCrabInt(CmpInst &I, crabLitFactory &lfac, const bool isNegated);
static Optional<var_t> unsignedCmpInstToCrabInt(CmpInst &I, crabLitFactory &lfac, const bool isNegated);
```
Then, when we call originally to `cmpInstToCrabInt(I, lfac, isNegated)` replace it with
```
if (I.isUnsigned()) {
  Optional<var_t> v = unsignedCmpInstToCrabInt(I, lfac, isNegated);
  if (vOpt.hasValue()) {
    // use vOpt.getValue()
  }
} else {
  Optional<lin_cst_t> cstOpt = signedCmpInstToCrabInt(I, lfac, isNegated);
  if (cstOpt.hasValue()) {
     // use cst_opt.getValue();
  }

}
```